sdks/java/testing/nexmark/src/main/java/org/apache/beam/sdk/nexmark/queries/Query3.java

```diff
@@ -164,21 +165,19 @@ public void processElement(ProcessContext c) {
     @StateId(PERSON)
     private static final StateSpec<ValueState<Person>> personSpec = StateSpecs.value(Person.CODER);
 
-    private static final String PERSON_STATE_EXPIRING = "personStateExpiring";
+    private static final String STATE_EXPIRING = "stateExpiring";
 
     @StateId(AUCTIONS)
-    private final StateSpec<ValueState<List<Auction>>> auctionsSpec =
-        StateSpecs.value(ListCoder.of(Auction.CODER));
+    private final StateSpec<BagState<Auction>> auctionsSpec = StateSpecs.bag(Auction.CODER);
 
-    @TimerId(PERSON_STATE_EXPIRING)
-    private final TimerSpec timerSpec = TimerSpecs.timer(TimeDomain.EVENT_TIME);
+    @TimerId(STATE_EXPIRING)
```

Comment:
Why are you making the expiration timer a processing-time timer? Isn't event time more correct here?

Reply:
Changed back to event time timer.